src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -674,6 +815,30 @@ til::point Terminal::_ConvertToBufferCell(const til::point viewportPos) const
     return bufferPos;
 }
 
+// Method Description:
+// - if necessary, scroll the viewport such that the given point is visible
+// Arguments:
+// - pos: a coordinate relative to the buffer (not viewport)
+void Terminal::_ScrollToPoint(const til::point pos)
+{
+    if (const auto visibleViewport = _GetVisibleViewport(); !visibleViewport.IsInBounds(pos))
+    {
+        if (const auto amtAboveView = visibleViewport.Top() - pos.Y; amtAboveView > 0)
+        {
+            // anchor is above visible viewport, scroll by that amount
+            _scrollOffset += amtAboveView;
+        }
+        else
+        {
+            // anchor is below visible viewport, scroll by that amount
+            const auto amtBelowView = pos.Y - visibleViewport.BottomInclusive();
+            _scrollOffset -= amtBelowView;
+        }
+        _NotifyScrollEvent();
+        _activeBuffer().TriggerScroll();

```

Comment:
Is it right to tell the text buffer that scrolling happened? Does this kick the renderer?

Reply:
Yup.
- `_NotifyScrollEvent()` --> tell TermControl to update the scroll bar
- `TriggerScroll()` --> tell the renderer to re-render the contents